### PR TITLE
feat: Add Typescript support.

### LINKED ISSUE #2: [NEEDED] index.d.ts
I've never done one of these before, despite seeing them in many NPM modules. This would be useful for TypeScript developers who would like to use `weeby-js` in TypeScript. Currently, this module is only fully supported for JavaScript.

### PR DESCRIPTION
**Pull request by WiseDevHelper**
Adds support for TypeScript developers.
Closes #2

### CODE REVIEW

File: typings/src/lib/generators.d.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        tweet: string;
+    }): Promise<any>;
+    /**
+      * Generates a Spotify Banner of statistics and information from either music, playlists or users. Please see https://weebyapi.xyz/api/docs#generators for the available generators.
+      * @param {Object} options - The options that contain the required parameters.
+      * @param {('album'|'artist'|'playlist'|'track'|'user')} options.type - The type of Spotify Search. (album, artist, playlist, track or user)
+      * @param {string} options.query - The value to search up. (Use %20 for spaces)
+      * @returns {Promise<buffer>} The generated image in a buffer.
+      */
+    spotifySearch(options?: {
```
Comment (NTMNathan) [inline typings/src/lib/generators.d.ts:144]: Spotify Search Endpoint was permanently removed due to issues with spotify api

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "include": [
+        "src/**/*.js",
+        "index.js"
+    ],
+    "compilerOptions": {
+        "target": "es2019",
```
Comment (NTMNathan) [inline tsconfig.json:7]: Just a suggestion, but I think use the latest compiler version? That is completely up to you

File: typings/src/lib/generators.d.ts (hunk 1 above)
Comment (WiseDevHelper) [reply typings/src/lib/generators.d.ts:144]: This still exists in the JS code?
https://github.com/ntm-development/weeby-js/blob/ee8202613c2a1b287594c8dbee4e6ec43e78a8d1/src/lib/generators.js#L314-L341

Comment (WiseDevHelper) [reply typings/src/lib/generators.d.ts:144]: Should this be removed too?

Comment (NTMNathan) [reply typings/src/lib/generators.d.ts:144]: The next update for Weeby-JS will have features added from the recent Weeby API update. So yes, Spotify Search will not be needed anymore

**Comment by NTMNathan** [comment]
Refer to the review comments i've made, but I think this LGTM 👍

**Review by NTMNathan — APPROVED** [review]
Looks great to me. I'll approve this and merge the changes 😃

**Pull request merged**